server/user.go

```diff
@@ -295,18 +313,30 @@ func (s *UserMgmtServer) getCreds(r *http.Request, requiresAdmin bool) (api.Cred
 		return api.Creds{}, err
 	}
 
-	isAdmin, err := s.cm.IsDexAdmin(clientID)
-	if err != nil {
-		log.Errorf("userMgmtServer: GetCreds err: %q", err)
-		return api.Creds{}, err
+	i := 0
+	for _, clientID := range clientIDs {
+		// Make sure the client actually exists.
+		isAdmin, err := s.cm.IsDexAdmin(clientID)
+		if err != nil {
+			log.Errorf("userMgmtServer: GetCreds err: %q", err)
+			return api.Creds{}, err
+		}
+		// If the endpoint requires an admin client, filter out clients which are not admins.
+		if isAdmin || !requiresAdmin {
+			clientIDs[i] = clientID
+			i++
+		}
 	}
-	if requiresAdmin && !isAdmin {
+
+	clientIDs = clientIDs[:i]
```

Comment:
The clients themselves can't be expired, only the token which holds a client_id. You can read some details on that token here[0].

Basically, the API gets that token and needs to verify the signature, make sure it hasn't expired, etc. _Normally_ it would also verify that the client_id is for the client we expect, that's why all the current verification methods take a client_id. However, for this this case we don't need to verify a token was issued for a specific client, we just need to check if that client is an admin later on (the line we're having this conversation about), which is done outside this verify call.

Actually the 278 shouldn't be a for loop, probably just a single verification call. The method just needs the client_id being passed is just verifying that the `aud` claim is for a client you expect. 

[0] https://openid.net/specs/openid-connect-core-1_0.html#IDToken